server.go

```diff
@@ -668,6 +671,16 @@ func (s *Server) applyCreateDatabase(m *messaging.Message) (err error) {
 	db := newDatabase()
 	db.name = c.Name
 
+	if s.RetentionAutoCreate {
+		// Create the default retention policy.
+		db.policies[c.Name] = &RetentionPolicy{
+			Name:     "default",
+			Duration: 0,
+			ReplicaN: 1,
+		}
```

Comment:
Also need to set it as default... db.defaultRetentionPolicy = "default".  It would be good to test for this but I don't think we currently have a way to query for the default policy name.